plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -104,7 +105,7 @@
     private final MachineRuntimeInfoImpl                      machineRuntime;
 
     @Inject
-    public DockerInstance(DockerConnector docker,
+    public DockerInstance(DockerConnectorProvider dockerProvider,
```

Comment:
Not sure I got the comment :( We do need to inject provider because now it is in charge of providing suitable connector (DockerConnector / OpenShiftConnector) based on `che.docker_connector.provider` property